multimedia/ffmpeg-devel/Portfile

```diff
@@ -183,11 +181,17 @@ platform darwin {
         configure.args-replace --disable-audiotoolbox --enable-audiotoolbox
     }
 
-    # libsdl2 requires minimum Xcode 10.7 SDK to build successfully
-    # but builds on Snow Leopard
     if {${os.major} > 9} {
+        # libsdl2 requires minimum Xcode 10.7 SDK to build successfully
+        # but builds on Snow Leopard
         configure.args-replace --disable-sdl2 --enable-sdl2
         depends_lib-append     port:libsdl2
+
+        # Dav1d: 10.6+ No PowerPC or Apple Silicon (for now)
+        if {[lsearch "arm powerpc" ${os.arch}] < 0} {
```

Comment:
Using the `in` or `ni` (not in) operator is simpler than `lsearch`ing.

But this isn't really the correct condition. Building this universal for arm64 x86_64 on arm64 would succeed but building it universal for arm64 x86_64 on x86_64 would fail. You should be looking at `[get_canonical_archs]` instead: if either arm64 or ppc or ppc64 are in that, then you can't add the dav1d dependency—if it doesn't support those archs (and it doesn't say it doesn't).

Reply:
perhaps just make it a bit simpler for now, and use ```dav1d``` if the os.major > 9, and forget about testing the arch? I think ```dav1d``` does build on ARM, and if it doesn't build on arm today, it likely will soon. And (AFAICT) nobody builds PowerPC code on >10.5  so we're good there.

Sure be nice to abstract this logic out of the ports that use a library, and keep it in the library itself, wouldn't it? You could have something like:
```
if {[build.supports davi1d]} {
    depends_lib-append     port:dav1d
    configure.args-append --enable-libdav1d
}
```
and leave the logic about whether the build (arch and OS) can support ```dav1d``` in ```david```.

Reply:
> @ryandesign wrote:
> You should be looking at `[get_canonical_archs]` instead: if either arm64 or ppc or ppc64 are in that, then you can't add the dav1d dependency—if it doesn't support those archs (and it doesn't say it doesn't).

Thanks! So, like this then?

```tcl
if {![regexp -all (arm64|ppc*) [get_canonical_archs]]} {
}
```
And I guess we won't have to check for `${os.major}` with that.

—


> @kencu wrote:
> You could have something like:
> 
> ```tcl
> if {[build.supports davi1d]} {
> }
> ```

I like that one. Also the naming… (for good readability). Clearly shows what's going on.